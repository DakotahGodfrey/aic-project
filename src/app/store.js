import { configureStore } from '@reduxjs/toolkit';
import homeReducer from 'app/slices/homeSlice';
import galleryReducer from 'app/slices/gallerySlice';
import artistsReducer from 'app/slices/artistsSlice';
import artworksReducer from 'app/slices/artworkSlice';
import artistReducer from 'app/slices/artistSlice';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    gallery: galleryReducer,
    artists: artistsReducer,
    artwork: artworksReducer,
    artist: artistReducer,
  },
});
