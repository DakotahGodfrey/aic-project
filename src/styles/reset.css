*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
}
html {
  font-size: 10px;
}
ol,
ul {
  list-style: none;
}
a {
  text-decoration: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
::-webkit-scrollbar {
  width: 0; /* Remove scrollbar space */
  background: transparent; /* Optional: just make scrollbar invisible */
}
