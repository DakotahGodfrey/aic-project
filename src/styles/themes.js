const defaultTheme = {
  textColor: '#222',
  secondaryColor: '#fff',
  navBG: '#111',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#666',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};
const darkTheme = {
  textColor: '#eee',
  navBG: '#111',
  secondaryColor: '#151515',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#ddd',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};

const getFromTheme = (propName) => (props) => props.theme[propName];
export { defaultTheme, darkTheme, getFromTheme };
