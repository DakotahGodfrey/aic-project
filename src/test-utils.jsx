import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

function render(ui, { theme, store, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>{children}</Router>
      </ThemeProvider>
    </Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react';
// override React Testing Library's render with our own
export { render };
