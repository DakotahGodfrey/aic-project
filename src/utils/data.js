const siteLinks = [
  { display: 'gallery', path: '/gallery', id: 1 },
  { display: 'artists', path: '/artists', id: 2 },
  { display: 'about', path: '/about', id: 3 },
];

const mediaQueries = {
  xSmall: 450,
  small: 600,
  medium: 768,
  large: 1024,
};

export { siteLinks, mediaQueries };
