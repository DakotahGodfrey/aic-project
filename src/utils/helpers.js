const IMAGES_PREFIX = 'https://www.artic.edu/iiif/2/';
const IMAGES_SUFFIX = '/full/843,/0/default.jpg';

const formatImageUrl = (id) => `${IMAGES_PREFIX}${id}${IMAGES_SUFFIX}`;

const generatePlaceholderArray = (size = 3) => {
  const placeholder = {
    thumbnail: {
      alt_text: 'content is loading',
    },
  };
  return new Array(size).fill(placeholder);
};
const smoothScrollIntoViewById = (id) => {
  document.querySelector(`#${id}`).scrollIntoView({ behavior: 'smooth' });
};
export { formatImageUrl, generatePlaceholderArray, smoothScrollIntoViewById };
