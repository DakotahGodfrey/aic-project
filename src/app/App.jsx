import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { darkTheme, defaultTheme } from 'styles/themes';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useLocation,
} from 'react-router-dom';
import SiteNav from 'components/layout/navigation/SiteNav';
import { siteLinks } from 'utils/data';
import Home from 'components/screens/Home';
import Gallery from 'components/screens/Gallery';
import Artists from 'components/screens/Artists';
import Artwork from 'components/screens/Artwork';
import Artist from 'components/screens/Artist';
import AboutApp from 'components/screens/AboutApp';
import NotFound from 'components/screens/NotFound';
import ScrollUpButton from 'components/layout/buttons/ScrollUpButton';

function ScrollToTop() {
  const { pathname } = useLocation();

  useEffect(() => {
    if (window instanceof Window) {
      window.scrollTo(0, 0);
    }
  }, [pathname]);

  return null;
}

const App = () => {
  const [theme, setTheme] = useState('default');

  const handleThemeToggle = () => {
    if (theme === 'default') {
      setTheme('darkTheme');
    } else {
      setTheme('default');
    }
    console.log(theme);
  };
  return (
    <ThemeProvider theme={theme === 'default' ? defaultTheme : darkTheme}>
      <Router>
        <div
          id='app'
          style={
            theme === 'default'
              ? { backgroundColor: '#eee' }
              : { backgroundColor: '#111' }
          }
        >
          <ScrollToTop />
          <SiteNav
            links={siteLinks}
            toggleTheme={handleThemeToggle}
            theme={theme}
          />
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={'/gallery'} component={Gallery} />
            <Route exact path={'/artists'} component={Artists} />
            <Route exact path={'/gallery/artwork/:id'} component={Artwork} />
            <Route exact path={'/artists/:id'} component={Artist} />
            <Route exact path={'/about'} component={AboutApp} />
            <Route component={NotFound} />
          </Switch>
          <ScrollUpButton />
        </div>
      </Router>
    </ThemeProvider>
  );
};

export default App;
