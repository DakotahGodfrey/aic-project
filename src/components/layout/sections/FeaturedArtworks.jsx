import { GridWrapper, GridCard } from 'components/layout/Layout';
import { Link } from 'react-router-dom';
import { formatImageUrl } from 'utils/helpers';
import placeholder from 'images/placeholder.svg';
import { BouncyEllipsis } from '../loaders';

const FeaturedArtworks = ({ artworks, columns = 3 }) => {
  return (
    <GridWrapper
      columns={columns}
      css={`
        padding-top: 1.5rem;
      `}
    >
      {artworks.map((artwork, index) =>
        artwork.id ? (
          <Link to={`/gallery/artwork/${artwork.id}`}>
            <GridCard key={artwork.id}>
              <img
                src={
                  artwork.image_id
                    ? formatImageUrl(artwork.image_id)
                    : placeholder
                }
                alt={
                  artwork.thumbnail ? artwork.thumbnail.alt_text : artwork.title
                }
              />
              <figcaption>
                <p className='title'>{artwork.title}</p>
                <p className='secondary'>
                  {artwork.artist_title ? artwork.artist_title : 'Unknown'}
                </p>
                <p className='secondary'>{artwork.medium_display}</p>
                <p className='secondary'>{artwork.date_display}</p>
              </figcaption>
            </GridCard>
          </Link>
        ) : (
          <GridCard key={index}>
            <img
              src={
                artwork.image_id
                  ? formatImageUrl(artwork.image_id)
                  : placeholder
              }
              alt={artwork.alt_text}
            />
            <figcaption>
              <BouncyEllipsis />
            </figcaption>
          </GridCard>
        )
      )}
    </GridWrapper>
  );
};

export default FeaturedArtworks;
