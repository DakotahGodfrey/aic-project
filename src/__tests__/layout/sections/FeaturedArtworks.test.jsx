import { store } from 'app/store';
import FeaturedArtworks from 'components/layout/sections/FeaturedArtworks';
import { defaultTheme } from 'styles/themes';
import { render, screen } from 'test-utils';
import placeholder from 'images/placeholder.svg';
import { generatePlaceholderArray } from 'utils/helpers';
const artworks = [
  {
    id: 'a',
    title: 'A',
    image_id: 'absasdfsd',
    thumbnail: {
      alt_text: 'test',
    },
    artist_title: 'testArtistA',
    medium_display: 'testMediumA',
    date_display: 'testDateA',
  },
  {
    id: 'b',
    title: 'B',
    image_id: 'absasdfsd',
    thumbnail: {
      alt_text: 'test',
    },
    artist_title: 'testArtistB',
    medium_display: 'testMediumB',
    date_display: 'testDateB',
  },
];

describe('<FeaturedArtworks />', () => {
  it.skip('throws an error if no featured artworks are provided', () => {
    expect(() => {
      render(<FeaturedArtworks />, { theme: defaultTheme, store: store });
    }).toThrow();
  });

  it('matches the snapshot', () => {
    const { container } = render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(container).toMatchSnapshot();
  });

  it('renders a link to the /gallery/artwork/:id route for each artwork in artworks prop', () => {
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    expect(screen.getAllByRole('link')).toHaveLength(artworks.length);
    for (let index = 0; index < artworks.length; index++) {
      expect(screen.getAllByRole('link')[index]).toHaveAttribute(
        'href',
        `/gallery/artwork/${artworks[index].id}`
      );
    }
  });
  it('renders an img element with the src set to url containing the artwork image_id', () => {
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    for (let index = 0; index < artworks.length; index++) {
      expect(screen.getAllByRole('img')[index].src).toContain(
        artworks[index].image_id
      );
    }
  });
  it('renders an img with correct alt_text', () => {
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    for (let index = 0; index < artworks.length; index++) {
      expect(screen.getAllByRole('img')[index].alt).toContain(
        artworks[index].thumbnail.alt_text
      );
    }
  });
  it('if no alt_text is provided it uses the artwork title as alt text', () => {
    artworks[0].thumbnail = '';
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    expect(screen.getAllByRole('img')[0].alt).toContain(artworks[0].title);
  });
  it('renders an element with the text content set to the artworks title', () => {
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    for (let index = 0; index < artworks.length; index++) {
      expect(screen.getAllByText(artworks[index].title)).toHaveLength(1);
    }
  });
  it('renders an element with the text content set to the artworks artist_title', () => {
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });
    for (let index = 0; index < artworks.length; index++) {
      expect(screen.getAllByText(artworks[index].artist_title)).toHaveLength(1);
    }
  });
  it('renders an element with the text content set to the artworks medium_display', () => {
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    for (let index = 0; index < artworks.length; index++) {
      expect(screen.getAllByText(artworks[index].medium_display)).toHaveLength(
        1
      );
    }
  });
  it('renders an element with the text content set to the artworks date_display', () => {
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    for (let index = 0; index < artworks.length; index++) {
      expect(screen.getAllByText(artworks[index].date_display)).toHaveLength(1);
    }
  });
  it('renders a placeholder if no image_id is present', () => {
    artworks[0].image_id = '';
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    expect(screen.getAllByRole('img')[0].src).toContain(placeholder);
  });
  it('renders a placeholder if artworkData is not present', () => {
    artworks[2] = {
      alt_text: 'testing',
    };
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getAllByRole('img')[2].src).toContain(placeholder);
  });
  it('renders a placeholders alt_text', () => {
    artworks[2] = {
      alt_text: 'testing',
    };
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    expect(screen.getAllByRole('img')[2].alt).toContain('testing');
  });
  it('renders the right placeholder text', () => {
    render(<FeaturedArtworks artworks={artworks} />, {
      theme: defaultTheme,
      store: store,
    });

    expect(screen.getByText(/loading/i)).toBeTruthy();
  });
});
