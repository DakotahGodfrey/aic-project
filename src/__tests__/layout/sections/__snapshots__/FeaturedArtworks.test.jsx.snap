// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedArtworks /> matches the snapshot 1`] = `
<div>
  <section
    class="sc-bdnxRM faqwCb"
  >
    <a
      href="/gallery/artwork/a"
    >
      <figure
        class="sc-gtsrHT bbAVtq"
      >
        <img
          alt="test"
          src="https://www.artic.edu/iiif/2/absasdfsd/full/843,/0/default.jpg"
        />
        <figcaption>
          <p
            class="title"
          >
            A
          </p>
          <p
            class="secondary"
          >
            testArtistA
          </p>
          <p
            class="secondary"
          >
            testMediumA
          </p>
          <p
            class="secondary"
          >
            testDateA
          </p>
        </figcaption>
      </figure>
    </a>
    <a
      href="/gallery/artwork/b"
    >
      <figure
        class="sc-gtsrHT bbAVtq"
      >
        <img
          alt="test"
          src="https://www.artic.edu/iiif/2/absasdfsd/full/843,/0/default.jpg"
        />
        <figcaption>
          <p
            class="title"
          >
            B
          </p>
          <p
            class="secondary"
          >
            testArtistB
          </p>
          <p
            class="secondary"
          >
            testMediumB
          </p>
          <p
            class="secondary"
          >
            testDateB
          </p>
        </figcaption>
      </figure>
    </a>
  </section>
</div>
`;
